numba/core/typing/templates.py

```diff
@@ -876,6 +885,26 @@ def get_template_info(self):
         }
         return info
 
+    @functools.lru_cache()
+    def get_cache_deps_info(self, sig):
+        """ return a dictionary with information about other functions this
+        function depends on
+
+        :param sig: signature
+        :return: dictionary of filenames and file stamps
+        """
+        from numba.core.caching import get_function_dependencies
```

Comment:
Rather than creating a dependency from caching in this file. Could we refactor the code to return the list of overloads, or the required pieces, and do the heavy lift in `caching.py`?

Reply:
fixed